### PR TITLE
Restructure and refine download package

### PR DESCRIPTION
**Pull request by seemantk**
@curran this PR is ready for your review (and handover)
The purpose: to restructure the downloads into the directory structure we discussed on today's call and generate a new download package:

`downloads/build` contains 4 dirs named after the sections.  Each (well, 3, because none exists for contribution-limits) of those now contains a file called `field-descriptions.csv`.

Similarly, I've restructured the `_modals` directory to have dirs named after the sections.  Each of those dirs contains a file called `about.md`, which is the raw markdown file with the modal content. So, I think those files can be copied over in the packaging script instead.

One consideration is that those raw .md files have YAML front-matter in them.  You can parse those out (everything between '---' and '---') or leave them in, as per your discretion.

### CODE REVIEW

**Comment by seemantk** [comment]
I had created a sub-branch for the cleanup stuff, and merged with that branch before this push. Sorry for the noise therein, I thought it would give more of a summary of that merge.

File: js/schematic.js [hunk 1]
```diff
@@ -298,22 +298,18 @@
...
       .domain(liquidToArray('{{ labels }}'))
-      .range(liquidToArray('{{ colors }}'))
-  ;
-  {% endif %}
-{% endfor %}{% endfor %}
+  {% endif %};
+  {% if legend.type == "threshold" %}colorScale["{{ section[0] }}"]["{{ legend.name }}"].emptyValue = {{ legend.fallback }};{% endif %}
+  {% endfor %}
+{% endfor %}
+  console.log(colorScale);
```
Comment (curran) [inline js/schematic.js]: @seemantk I'd appreciate it very much if you could please make separate PRs for refactoring and features. As it is here, I'm seeing lots of changes not related to the download feature, so I need to go through and verify all other features that may be affected by these refactoring changes in addition to verifying the download changes.

**Comment by seemantk** [comment]
Hi Curran,

That's actually what I tried here, but the branch off the branch didn't
show separately when I pushed.

Sorry for the inconvenience, I'm learning github workflow as we go.

Cheers,
Seemant

**Comment by curran** [comment]
@seemantk Aha got it - yeah I hate that, how the PR always shows the diff from master not the diff from a previous PR. Sorry, my misunderstanding.

Maybe it would work out differently if the work were first done on the feature, then refactoring, rather than the other way around (at least from a workflow perspective).

File: css/vendors.min.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+# We could source thse via CDN instead.
```
Comment (curran) [inline css/vendors.min.css]: Good idea! I like unpkg.

**Comment by curran** [comment]
It looks like something is not quite right with the Jekyll build, all the CSV files have the header row only. Here's what I see:

```
curran@neon:~/repos/cfi/downloads$ jekyll build
Configuration file: none
            Source: /home/curran/repos/cfi/downloads
       Destination: /home/curran/repos/cfi/downloads/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
                    done in 0.016 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
curran@neon:~/repos/cfi/downloads$ cd _site/
curran@neon:~/repos/cfi/downloads/_site$ tree
.
├── build
│   ├── disclosure
│   │   └── field-descriptions.csv
│   ├── other-restrictions
│   │   └── field-descriptions.csv
│   └── public-financing
│       └── field-descriptions.csv
└── cfi-laws-database.zip

4 directories, 4 files
curran@neon:~/repos/cfi/downloads/_site$ cat build/disclosure/field-descriptions.csv 
Field Name,Question,Short Label,Additional Notes
curran@neon:~/repos/cfi/downloads/_site$ cat build/other-restrictions/field-descriptions.csv 
Field Name,Question,Short Label,Additional Notes
curran@neon:~/repos/cfi/downloads/_site$ cat build/public-financing/field-descriptions.csv 
Field Name,Question,Short Label,Additional Notes
curran@neon:~/repos/cfi/downloads/_site$ 
```

**Comment by seemantk** [comment]
oi! I'll have a look too.

**Comment by curran** [comment]
Got it! Sorry, a stupid mistake on my end. I ran `jekyll build` from within `cfi/downloads`, not `cfi`.

I just now discovered this and now I see the CSV content filled in properly. Sorry to cause alarm :smile:

**Comment by curran** [comment]
Aha - the task before me is to figure out which CSV files contain the fields in each section directory. I'm on it.

**Comment by curran** [comment]
At this point we've got the following directory structure in the .zip file:

```
cfi-laws-database/
├── contribution-limits
│   ├── Laws_02_Contributions_1.csv
│   ├── Laws_02_Contributions_2.csv
│   └── Laws_02_Contributions_3.csv
├── disclosure
│   ├── field-descriptions.csv
│   ├── Laws_03_Disclosure_1.csv
│   ├── Laws_03_Disclosure_2.csv
│   └── Laws_03_Disclosure_3.csv
├── Laws_00_IdentifierTable.csv
├── Laws_01_Defintions.csv
├── LICENSE
├── other-restrictions
│   ├── field-descriptions.csv
│   └── Laws_05_Other.csv
└── public-financing
    ├── field-descriptions.csv
    └── Laws_04_PublicFinancing.csv
```

**Comment by curran** [comment]
After adding the about buttons modal markdown content, here's what we have:

```
cfi-laws-database
├── contribution-limits
│   ├── about.md
│   ├── howto.md
│   ├── Laws_02_Contributions_1.csv
│   ├── Laws_02_Contributions_2.csv
│   └── Laws_02_Contributions_3.csv
├── disclosure
│   ├── about.md
│   ├── field-descriptions.csv
│   ├── howto.md
│   ├── Laws_03_Disclosure_1.csv
│   ├── Laws_03_Disclosure_2.csv
│   └── Laws_03_Disclosure_3.csv
├── Laws_00_IdentifierTable.csv
├── Laws_01_Defintions.csv
├── LICENSE
├── other-restrictions
│   ├── about.md
│   ├── field-descriptions.csv
│   ├── howto.md
│   └── Laws_05_Other.csv
└── public-financing
    ├── about.md
    ├── field-descriptions.csv
    ├── howto.md
    └── Laws_04_PublicFinancing.csv
```

@seemantk As for the front-matter YML, ideally we could keep the title as a Markdown h1, so this

```
---
layout: modal
modal: about-disclosure
title: About Disclosure
---
```

could become this

```
# About Disclosure
```

One easy way to do this would be to add it to the markdown, and remove the bit of JS that renders the heading (I think this is how we do it... will need to take a look).

**Comment by curran** [comment]
Stripped out the frontmatter. Created a new issue for addressing the title issue later **Include titles in MD files in download package** #188.

@seemantk I think this is good to merge now.
